var/spack/repos/builtin/packages/parsec/package.py

```diff
@@ -104,15 +104,25 @@ def check(self):
                 warn += "https://bitbucket.org/icldistcomp/parsec/issues"
                 tty.msg(warn)
 
-    def test(self):
+    def test_parsec(self):
         """Compile and run a user program with the installed library"""
         with working_dir(join_path(self.install_test_root, "contrib/build_with_parsec")):
-            self.run_test(
-                "cmake", options=["."], purpose="Check if CMake can find PaRSEC and its targets"
-            )
-            self.run_test("make", purpose="Check if tests can compile")
-            self.run_test("./dtd_test_allreduce")
-            self.run_test("./write_check")
+
+            with test_part(
+                self, "test_parsec_cmake", purpose="Check if CMake can find PaRSEC and its targets"
+            ):
+                cmake = which("cmake")
+                cmake(".")
+
+            with test_part(self, "test_parsec_make", purpose="Check if tests can compile"):
+                make = which("make")
+                make()
```

Comment:
One of the goals of the new API is to simplify/tighten stand-alone test implementations s.t. they are more in line with build recipes.  To that end, it is no longer desirable to have separate test parts for `cmake` and `make` commands that are used to build test programs.

Reply:
Recall the structure of the example at https://spack.readthedocs.io/en/latest/packaging_guide.html#saving-build-and-install-time-files.

Reply:
Also, recall you want to use the version of `cmake` associated with the build (i.e., self.spec["cmake"].command).